Objects/methodobject.c

```diff
@@ -173,12 +173,16 @@ meth_dealloc(PyObject *self)
     if (m->m_weakreflist != NULL) {
         PyObject_ClearWeakRefs((PyObject*) m);
     }
+    // We need to access ml_flags here rather than later.
+    // `m` might have the same lifetime
```

Comment:
m->ml I think here, but yes.

You are also right that probably we should update the API documentation to clarify this is an acceptable thing to do.